### ISSUE TITLE
WGAN experiences

### ISSUE THREAD

**Issue by kadarakos** [jerryli27/my_pix2pix#1]
I'm using the pix2pix for a project I'm working on https://github.com/tcsai/tcsai.github.io .
I have implemented the WGAN independently of you, but then I've found your implementation,
which seemed pretty helpful, so now the code looks exactly like yours.

However, the WGAN doesn't seem to learn to generate as good images as the default.
What are your experiences? Did you get it to work?

**Comment by jerryli27** [comment]
Hi. Sorry for replying so late. I don't usually check my repo because I wasn't expecting others to use it. Regarding your question, I've found a way for wgan to work. If you compare https://github.com/jerryli27/my_pix2pix/blob/master/pix2pix_w_hint_lab_wgan.py with https://github.com/jerryli27/my_pix2pix/blob/master/pix2pix_w_hint_512.py and only focus on the changes I've made in the loss function and the discriminator, you should see how it's implemented. I've tried it on my own dataset and it indeed increased the model's variety (meaning it outputs more types of correct output) Some minor things that is different from the paper is; I don't find training the discriminator extra rounds help at all. It only slows down the training process. (But that could be I was not implementing that part correctly). Hope this helps!

**Comment by jerryli27** [comment]
And let me know if you have any questions. Shoot me a reminder at jiamingli2017 (at) u.northwestern.edu. if I'm not responding on github.
